homeassistant/components/sensor/citybikes.py

```diff
@@ -0,0 +1,186 @@
+    PLATFORM_SCHEMA.extend({
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_NETWORK): cv.string,
+        vol.Inclusive(CONF_LATITUDE, 'coordinates'): cv.latitude,
+        vol.Inclusive(CONF_LONGITUDE, 'coordinates'): cv.longitude,
+        vol.Exclusive(CONF_RADIUS, 'station_filter'): cv.positive_int,
+        vol.Exclusive(CONF_STATIONS_LIST, 'station_filter'):
+            vol.All(
+                cv.ensure_list,
+                vol.Length(min=1),
+                [cv.string])
+    }))
+
+
+def _filter_stations(network, radius, latitude, longitude, stations_list):
+    if radius > 0:
+        for station, dist in network.stations.near(latitude, longitude):
+            # 'dist' is in weird units, let's calculate again
+            dist = location.distance(latitude, longitude,
+                                     station[ATTR_LATITUDE],
+                                     station[ATTR_LONGITUDE])
+            if dist > radius:
+                break
+            yield station
+    else:
+        for station in network.stations:
+            if station['id'] in stations_list:
+                yield station
+                continue
+            if 'extra' in station:
+                if 'uid' in station['extra']:
+                    if str(station['extra']['uid']) in stations_list:
+                        yield station
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the CityBikes platform."""
+    from citybikes import Client, Network
+
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+
+    network_uid = config.get(CONF_NETWORK)
+    if not network_uid:
+        network = Client().networks.near(latitude, longitude)[0][0]
+    else:
+        network = Network(Client(), uid=network_uid)
+
+    stations_list = config.get(CONF_STATIONS_LIST, [])
+    radius = config.get(CONF_RADIUS, 0)
+    if not hass.config.units.is_metric:
+        radius = distance.convert(radius, LENGTH_FEET, LENGTH_METERS)
+    poller = CityBikesNetworkPoller(network)
+
+    add_devices(CityBikesStationSensor(poller, station,
+                                       config.get(CONF_NAME, DOMAIN))
+                for station in _filter_stations(network, radius, latitude,
+                                                longitude, stations_list))
+
+
+class CityBikesNetworkPoller(object):
+    """A wrapper around a CityBikes Network object."""
+
+    def __init__(self, network):
+        """Initialize the poller."""
+        self._network = network
+
+    def get(self, station_id, update=True):
+        """Return the station with the given id."""
+        if update:
+            self._update()
+        for station in self._network.stations:
+            if station[ATTR_STATION_ID] == station_id:
+                return station
+
+    @Throttle(SCAN_INTERVAL)
```

Comment:
What's throttled here is the `update` method of the poller object, not the station entities. Every time a `SCAN_INTERVAL` passes, each monitored station will indirectly call upon the `update` method of the caller, and the `Throttle` decorator makes sure only one actual request to the API is made.

It would be cleaner if each station could request data about itself, but unfortunately, that's not currently supported by the API.

Another option would be to create a separate entity for the whole bike sharing network, which will be updated at `SCAN_INTERVAL`, and then manually update each station entity. I'm planning to do it that way later, so that the platform is set up correctly when HASS starts offline, and the stations are added later, once connectivity is restored. But for now, I'm not sure how to achieve that correctly, hence the current approach.